libs/BLEKit/tests/CoreGattServerEventHandler_test.cpp

```diff
@@ -0,0 +1,154 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#include "CoreGattServerEventHandler.h"
+
+#include "ble_mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "mocks/leka/BLEService.h"
+
+using namespace leka;
+using namespace ble;
+
+using ::testing::InSequence;
+using ::testing::MockFunction;
+using ::testing::Return;
+
+class CoreGattServerEventHandlerTest : public testing::Test
+{
+  protected:
+	void SetUp() override
+	{
+		core_gatt_server_event_handler.registerServerWrite(mock_server_write_func.AsStdFunction());
+		core_gatt_server_event_handler.setServices(_services);
+	}
+	void TearDown() override { ble::delete_mocks(); }
+
+	BLE &ble = BLE::Instance();
+	CoreGattServerEventHandler core_gatt_server_event_handler {};
+
+	MockFunction<updateServiceFunction> mock_server_write_func;
+
+	const UUID _uuid {0x0001};
+	static const uint8_t _services_count {1};
+	static const uint8_t _characteristics_count {1};
+	uint8_t _characteristic_value {42};
+	ReadOnlyGattCharacteristic<uint8_t> _writable_characteristic {0x1234, &_characteristic_value};
```

Comment:
tu l'appelles writable alors qu'elle est read only.

```suggestion
	ReadOnlyGattCharacteristic<uint8_t> _characteristic {0x1234, &_characteristic_value};
```

Reply:
C'est corrigé